src/core/cgroup.c

```diff
@@ -1342,6 +1349,43 @@ static void cgroup_apply_firewall(Unit *u) {
         (void) bpf_firewall_install(u);
 }
 
+void cgroup_modify_nft_set(Unit *u, bool add) {
+        int r;
+        CGroupContext *c;
+
+        assert(u);
+
+        if (!MANAGER_IS_SYSTEM(u->manager))
```

Comment:
any specific reason this cannot be supported for user units?

Reply:
User units would be OK. But isn't this check whether this instance is PID1 or one of `systemd --user` instances?

Reply:
but then how it can work if this call is refused? Perhaps the user manager should proxy it back to the system manager if it's about privileges?

Reply:
More generally -- how does this work with `Delegate=yes`? A service may create arbitrary subtree with internal cgroup_ids that would be opaque to PID 1 handling NFT rules mapping (path <-> cgroup id).

Reply:
And on that topic, I think this should be supported at least for scopes too - maybe even slices. And what about sockets?

Reply:
Sorry, I didn't think my answer through. User manager can't add elements to NFT sets without CAP_NET_ADMIN. I wouldn't implement any proxying from unprivileged manager to PID1 as it would modify firewall configuration based on unprivileged users input. I don't think delegated subtrees would be OK either.

Slices and sockets do indeed work, I'm using both in my configuration, so I think scopes are fine too. It's very useful to make separate firewall chains for `system.slice` and `user.slice`, then for example `ssh.socket`.